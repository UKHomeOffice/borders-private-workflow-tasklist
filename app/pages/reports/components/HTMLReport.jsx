import React from 'react';
import PropTypes from 'prop-types';
import Iframe from 'react-iframe';

export default function HTMLReport({ htmlName, reportServiceUrl }) {
  return (
    <Iframe
      url={`${reportServiceUrl}/api/reports/${htmlName}`}
      id="report"
      width="100%"
      height="100%"
      position="relative"
      display="initial"
      allowFullScreen
    />
  );
}

HTMLReport.propTypes = {
  htmlName: PropTypes.string.isRequired,
  reportServiceUrl: PropTypes.string.isRequired,
};
