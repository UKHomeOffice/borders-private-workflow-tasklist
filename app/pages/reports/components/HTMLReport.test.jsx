import React from 'react';
import HTMLReport from './HTMLReport';

describe('HTMLReport', () => {
  const props = {
    reportServiceUrl: 'http://localhost:9000',
    htmlName: 'HTML Report',
  };

  it('renders an HTMLReport', () => {
    const wrapper = shallow(<HTMLReport {...props} />);
    expect(wrapper.find('Iframe').exists()).toEqual(true);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<HTMLReport {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
