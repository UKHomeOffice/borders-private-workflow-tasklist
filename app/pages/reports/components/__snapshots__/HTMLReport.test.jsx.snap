// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTMLReport matches snapshot 1`] = `
<Iframe
  allowFullScreen={true}
  display="initial"
  height="100%"
  id="report"
  position="relative"
  url="http://localhost:9000/api/reports/HTML Report"
  width="100%"
/>
`;
